crates/bls_utils/src/bls.rs

```diff
@@ -89,21 +104,26 @@ pub fn lagrange_interpolation(
     Ok(G1Affine::from(r))
 }
 
-pub fn hash_message_to_g2(msg: &[u8], domain: &[u8]) -> G2Projective {
-    <G2Projective as HashToCurve<ExpandMsgXmd<Sha256>>>::hash_to_curve([msg], domain)
-}
-
-pub fn bls_verify(pubkey: &G1Affine, signature: &G2Affine, message: &[u8]) -> bool {
+pub fn hash_message_to_g2(msg: &[u8]) -> G2Projective {
     let domain = b"BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_POP_";
-    let pk_projective = G1Projective::from(pubkey);
-    let sig_projective = G2Projective::from(signature);
+    <G2Projective as HashToCurve<ExpandMsgXmd<Sha256>>>::hash_to_curve(msg, domain)
```

Comment:
Interesting. I thought maybe you could drop `G2Projective as`, but it turns out that you cannot. Using `dyn` also does not help. So much for knowing Rust.